components/benefit_markets/app/models/benefit_markets/services/pvp_eligibility_service.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module BenefitMarkets
+  module Services
+    class PvpEligibilityService
+      attr_reader :product, :current_user
+
+      def initialize(product, current_user, args = {})
+        @product = product
+        @current_user = current_user
+        @args = args
+      end
+
+      def create_or_update_pvp_eligibilities
+        eligibility_result = {}
+
+        @args[:rating_areas].each do |rating_area_id, evidence_value|
+          pvp = find_or_create_pvp(rating_area_id)
+          effective_date = @args[:effective_date] || TimeKeeper.date_of_record
+          current_eligibility = pvp.latest_active_pvp_eligibility_on(effective_date)
+          next if current_eligibility.present?.to_s == evidence_value.to_s
+
+          result = store_pvp_eligibility(pvp, evidence_value, effective_date)
+          eligibility_result[rating_area_id] = result.success? ? "Success" : "Failure"
+        end
+
+        grouped_eligibilities = eligibility_result.group_by { |_year, value| value }
+        grouped_eligibilities.transform_values { |items| items.map(&:first) }
+      end
+
+      def find_or_create_pvp(rating_area_id)
```

Comment:
are this and the following method intended to be `private` perhaps?

No is fine, just thought it might clarify intent.